core/templates/dev/head/domain/collection/CollectionUpdateService.js

```diff
@@ -257,21 +279,30 @@ oppia.factory('CollectionUpdateService', [
         },
 
         /**
+         * Returns whether the collection tag is valid.
+         */
+        isTagValid: function(collection, tags) {
```

Comment:
Actually, this validation should go into `CollectionValidationService`. Sorry if I wrote the wrong thing earlier!